<?xml version="1.0" encoding="UTF-8"?>
<launch>
	<arg name="sigverse_ros_bridge_port"        default="50001" />
	<arg name="ros_bridge_port"                 default="9090" />

	<node pkg="sigverse_tiago_teleop_key" exec="tiago_teleop_key" output="screen" launch-prefix="gnome-terminal -t 'TIAGo teleop key' -- ">
		<!--
		<remap from="/tiago/message/to_robot"         to="/tiago/message/to_robot" />
		<remap from="/tiago/message/to_human"         to="/tiago/message/to_human" />
		<remap from="/joint_states"                   to="/joint_states" />
		<remap from="/mobile_base_controller/cmd_vel" to="/mobile_base_controller/cmd_vel" />
		<remap from="/torso_controller/command"       to="/torso_controller/command" />
		<remap from="/head_controller/command"        to="/head_controller/command" />
		<remap from="/arm_controller/command"         to="/arm_controller/command" />
		<remap from="/gripper_controller/command"     to="/gripper_controller/command" />
		-->
	</node>

	<node pkg="sigverse_ros_bridge" exec="sigverse_ros_bridge" args="$(var sigverse_ros_bridge_port)"/>

	<include file="$(find-pkg-share rosbridge_server)/launch/rosbridge_websocket_launch.xml" > 
		<arg name="port" value="$(var ros_bridge_port)"/>
		<arg name="default_call_service_timeout" value="5.0"/>
		<arg name="call_services_in_new_thread" value="true"/>
		<arg name="send_action_goals_in_new_thread" value="true"/>
<!--		<arg name="bson_only_mode" value="true"/> -->
	</include>
</launch>


